files/bootstrap.sh

```diff
@@ -32,6 +32,7 @@ function print_help {
     echo "--service-ipv6-cidr ipv6 cidr range of the cluster"
     echo "--enable-local-outpost Enable support for worker nodes to communicate with the local control plane when running on a disconnected Outpost. (true or false)"
     echo "--cluster-id Specify the id of EKS cluster"
+    echo "--imds-endpoint sets the IP address or hostname port pair for accessing the EC2 Metadata Service (default: 169.254.169.254:80)"
```

Comment:
Not sure I grepped the purpose of this in earlier revisions -- does this only exist so that the tests can set the endpoint to `localhost`? Or is there a use-case in a real environment?

Reply:
Only for testing. I assume you'd rather leave this out from args then and maybe only consume as an env var?

Reply:
actually now that I think about it, there is a use-case in a real environment if a user is running on an ipv6 only subnet. They could use this argument to specify the ipv6 EC2 IMDS endpoint

Reply:
Oh interesting, the IPv6 IMDS endpoint is only usable on Nitro 🤔 https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/configuring-instance-metadata-service.html

I'm not *super* concerned about adding this to the bootstrap contract; it should be dead simple to support in the future. But if there's a way we can just backdoor this for testing purposes, that's preferable IMO.

Reply:
sgtm, most instance types are nitro now, so that's not a huge issue FYI

Reply:
ack, I need to update my lingo 😂